spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-configuration-watcher-it/src/test/java/org/springframework/cloud/kubernetes/configuration/watcher/ActuatorRefreshMultipleNamespacesIT.java

```diff
@@ -0,0 +1,221 @@
+
+	private static final String DEFAULT_NAMESPACE = "default";
+
+	private static final String LEFT_NAMESPACE = "left";
+
+	private static final String RIGHT_NAMESPACE = "right";
+
+	private static final K3sContainer K3S = Commons.container();
+
+	private static Util util;
+
+	@BeforeAll
+	static void beforeAll() throws Exception {
+		K3S.start();
+		Commons.validateImage(SPRING_CLOUD_K8S_CONFIG_WATCHER_APP_NAME, K3S);
+		Commons.loadSpringCloudKubernetesImage(SPRING_CLOUD_K8S_CONFIG_WATCHER_APP_NAME, K3S);
+		util = new Util(K3S);
+		util.createNamespace(LEFT_NAMESPACE);
+		util.createNamespace(RIGHT_NAMESPACE);
+		util.setUpClusterWide(DEFAULT_NAMESPACE, Set.of(DEFAULT_NAMESPACE, LEFT_NAMESPACE, RIGHT_NAMESPACE));
+	}
+
+	@AfterAll
+	static void afterAll() throws Exception {
+		util.deleteNamespace(LEFT_NAMESPACE);
+		util.deleteNamespace(RIGHT_NAMESPACE);
+		Commons.cleanUp(SPRING_CLOUD_K8S_CONFIG_WATCHER_APP_NAME, K3S);
+	}
+
+	@BeforeEach
+	void setup() {
+		configWatcher(Phase.CREATE);
+		util.wiremock(DEFAULT_NAMESPACE, "/", Phase.CREATE);
+	}
+
+	@AfterEach
+	void after() {
+		configWatcher(Phase.DELETE);
+		util.wiremock(DEFAULT_NAMESPACE, "/", Phase.DELETE);
+	}
+
+	/**
+	 * <pre>
+	 *     - deploy config-watcher in default namespace
+	 *     - deploy wiremock in default namespace (so that we could assert calls to the actuator path)
+	 *     - deploy configmap-left in left namespaces with proper label and "service-wiremock" name. Because of the
+	 *       label, this will trigger a reload; because of the name this will trigger a reload against that name.
+	 *       This is a http refresh against the actuator.
+	 *     - same as above for the configmap-right.
+	 * </pre>
+	 */
+	@Test
+	void testConfigMapActuatorRefreshMultipleNamespaces() {
+		WireMock.configureFor(WIREMOCK_HOST, WIREMOCK_PORT, WIREMOCK_PATH);
+		await().timeout(Duration.ofSeconds(60))
+				.until(() -> WireMock
+						.stubFor(WireMock.post(WireMock.urlEqualTo("/actuator/refresh"))
+								.willReturn(WireMock.aResponse().withBody("{}").withStatus(200)))
+						.getResponse().wasConfigured());
+
+		// left-config-map
+		V1ConfigMap leftConfigMap = new V1ConfigMapBuilder().editOrNewMetadata()
+				.withLabels(Map.of("spring.cloud.kubernetes.config", "true")).withName("service-wiremock")
+				.withNamespace(LEFT_NAMESPACE).endMetadata().addToData("color", "purple").build();
+		util.createAndWait(LEFT_NAMESPACE, leftConfigMap, null);
+
+		// right-config-map
+		V1ConfigMap rightConfigMap = new V1ConfigMapBuilder().editOrNewMetadata()
+				.withLabels(Map.of("spring.cloud.kubernetes.config", "true")).withName("service-wiremock")
+				.withNamespace(RIGHT_NAMESPACE).endMetadata().addToData("color", "green").build();
+		util.createAndWait(RIGHT_NAMESPACE, rightConfigMap, null);
+
+		// comes from handler::onAdd (and as such from "onEvent")
+		Commons.assertReloadLogStatements("ConfigMap service-wiremock was added in namespace left", "",
```

Comment:
these two asserts are the main idea of the test: if they happen, it means we listen for changes in "selective namespaces" and its exactly what we want. This test will fail in `main` branch for example, because has this bug.